contracts/src/TeleporterTokenDestination.sol

```diff
@@ -195,81 +198,109 @@ abstract contract TeleporterTokenDestination is
      * - {input.destinationBridgeAddress} cannot be the zero address
      * - {input.recipient} cannot be the zero address
      * - {amount} must be greater than 0
-     * - {amount} must be greater than {input.primaryFee}
      */
     function _send(SendTokensInput calldata input, uint256 amount) internal sendNonReentrant {
-        require(input.recipient != address(0), "TeleporterTokenDestination: zero recipient address");
-        require(input.requiredGasLimit > 0, "TeleporterTokenDestination: zero required gas limit");
+        _validateSendTokensInput(input);
+        require(
+            input.destinationBlockchainID == sourceBlockchainID,
+            "TeleporterTokenDestination: invalid destination blockchain ID"
+        );
+        require(
+            input.destinationBridgeAddress == tokenSourceAddress,
+            "TeleporterTokenDestination: invalid destination bridge address"
+        );
```

Comment:
Feels a little odd to have input param values that we require to be known values. Could consider removing `destinationBlockchainID` and `destinationBridgeAddress` from input in this case, but then it wouldn't be symmetric with the input on the source contracts.

Reply:
ye if we're going to go down this route I'd be curious to explore changes to remove the not needed blockchain ID, bridge address for destination -> source. Will be a decent amount of refactoring with the interfaces though.